### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by priesgo**
* Adding option to disable annotations for everey operation
* Removing non consolidated references from output from webservices
* Annotation in batches of variants
* Cellbase timeout
* Transaction retry endpopint only for cancelled transactions
* Create mongo indices on startup for all fields that are queried
* VCF loaders refactored to annotate in batches + use a single jar called cva-app

@indikaraj

### CODE REVIEW

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        List<VariantAnnotation> variantAnnotations = null;
+
+        // annotates a list of variants
+        logger.info(String.format("Annotating %d variants...", variants.size()));
+        ExecutorService executorService = Executors.newFixedThreadPool(1);
+        Future<List<VariantAnnotation>> futureAnnotations = executorService.submit(() -> {
+                return cellBaseVariantAnnotator.annotate(variants);
+        });
+
+        try {
```
Comment (indikaraj) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java]: How about refactoring the code to use already provided future functionality. Added below for reference
try {
   futureAnnotations.get(timeOutThreshold, TimeUnit.SECONDS)
} catch(ExecutionException|TimeoutExeption|InterruptedException e) {
   futureAnnotations.cancel(true);
  throw new CvaException(e.getMessage)
}

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    private static Logger logger = CvaConfiguration.getLogger(CellbaseAnnotator.class);
+
+    public static List<Variant> annotateVariants(List<Variant> variants) throws CvaException {
+
+        AbstractCellBaseVariantAnnotator cellBaseVariantAnnotator = CvaConfiguration.getCellBaseVariantAnnotator();
+        List<VariantAnnotation> variantAnnotations = null;
+
+        // annotates a list of variants
+        logger.info(String.format("Annotating %d variants...", variants.size()));
+        ExecutorService executorService = Executors.newFixedThreadPool(1);
```
Comment (indikaraj) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java]: I think, there are few options here....
1. create a executor service with one thread for each request for cell base and shutdown it in a finally block once complete (once complete with a success or timeout or failure). Much safe approach. Not that expensive anyway
2. Have fixed thread pool with a reasonable size (for handling requests to cell base) for the duration of the application. (Thread pool size of 1 will not be enough as cancel operation after a timeout doesn't guarantee to return the control immediately)

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java (hunk 1 above)
Comment (priesgo) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java]: Done. Really useful!

File: cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java (hunk 2 above)
Comment (priesgo) [inline cva-storage/cva-storage-core/src/main/java/org/gel/cva/storage/core/helpers/CellbaseAnnotator.java]: done! I chose option 2 with a default number of 5 threads configurable using an environment variable.

**Pull request merged**